src/core/datastore/sync/sync-manager.js

```diff
@@ -60,6 +66,12 @@ export class SyncManager {
     if (!isNonemptyString(collection)) {
       return Promise.reject(new KinveyError('Invalid or missing collection name'));
     }
+
+    // TODO: decide on default value of pagination setting
+    if (options && (options.autoPagination && !options.useDeltaFetch)) {
```

Comment:
Using `useDeltaFetch` here seems incorrect. Whether to get the entire set of objects or a delta, should be decided based on the `useDeltaFetch` value and the `_QueryCache`. 

If I have run this query before (regardless of whether I ran it as "auto-paginated" or not) _and_ I have delta sync turned on, then make a delta request for me.

Reply:
Well yeah, but delta set isn't implemented here yet. This will have to be done after delta set is ready.

Reply:
Also, what's the decision on the setting - setting, no setting, default value?

Reply:
Can you explain what you mean "setting, no setting, default value"?

Reply:
There is a discussion in the second spec doc about whether auto pagination should be a setting or not. And if it is a setting, what should be its default value? There is no final decision yet. I've left it as a seeting, with a default value of `false`, as per the original spec doc.